src/CodeCoverage.php

```diff
@@ -861,7 +859,11 @@ private function initializeData(): void
         $this->isInitialized = true;
 
         if ($this->processUncoveredFilesFromWhitelist) {
-            $this->shouldCheckForDeadAndUnused = false;
+
+            // by collecting dead code data here on an initial pass, future runs with test data do not need to
+            if ($this->driver->canDetectDeadCode()) {
```

Comment:
This optimisation function relies on dead code detection being enabled so I've made enabling this explicit rather than just relying on the default value of the driver